### PR TITLE
feat(metadata): add vulnerability-database-updated-at in metadata

### PR DESCRIPTION
**Pull request by heww**
returns `harbor.scanner-adapter/vulnerability-database-updated-at` in metadata when config clair database url for the scanner

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ Configuration of the adapter is done via environment variables at startup.
 | `SCANNER_API_SERVER_WRITE_TIMEOUT` | `15s` | The maximum duration before timing out writes of the response. |
 | `SCANNER_API_SERVER_IDLE_TIMEOUT`  | `60s` | The maximum amount of time to wait for the next request when keep-alives are enabled. |
 | `SCANNER_CLAIR_URL`                | `http://harbor-harbor-clair:6060` | Clair URL |
+| `SCANNER_CLAIR_DATABASE_URL`       | | The Clair database URL, it is used to fetch vulnerability database updated time of the Clair. Its format is `postgresql://user:password@host/db?sslmode=disable` |
```
Comment (steven-zou) [inline README.md]: `| |` two columns?

File: pkg/clair/client.go [hunk 2]
```diff
@@ -3,29 +3,51 @@ package clair
...
+	db          *sql.DB
 	endpointURL string
 	// need to customize the logger to write output to job log.
 	client *http.Client
 }
 
 // NewClient constructs a new client for Clair REST API pointing to the specified endpoint URL.
 func NewClient(tlsConfig etc.TLSConfig, cfg etc.ClairConfig) Client {
+	var db *sql.DB
+	if cfg.DatabaseURL != "" {
```
Comment (steven-zou) [inline pkg/clair/client.go]: I don't think we should return an error here as CVE timestamp is an optional feature. Logging it is ok.

File: pkg/http/api/v1/handler_test.go [hunk 3]
```diff
@@ -3,21 +3,23 @@ package v1
...
-	"github.com/goharbor/harbor-scanner-clair/pkg/job"
 	"net/http"
 	"net/http/httptest"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/goharbor/harbor-scanner-clair/pkg/harbor"
+	"github.com/goharbor/harbor-scanner-clair/pkg/job"
+
```
Comment (steven-zou) [inline pkg/http/api/v1/handler_test.go]: extra line

File: README.md (hunk 1 above)
Comment (heww) [reply README.md]: There are three columns in the table, the middle one is the default value of the config, for `SCANNER_CLAIR_DATABASE_URL` is empty string which means not return the vulnerability database updated time in the metadata of the scanner.

File: pkg/http/api/v1/handler_test.go (hunk 3 above)
Comment (heww) [reply pkg/http/api/v1/handler_test.go]: fixed

File: pkg/clair/client.go (hunk 2 above)
Comment (heww) [reply pkg/clair/client.go]: this feature enabled only clair database url configed, return error is ok when users enable this feature but adapter can not connect the db

**Comment by steven-zou** [comment]
as we both approved this PR, let's merge it.

**Pull request merged**